pkg/flow/internal/controller/config.go

```diff
@@ -0,0 +1,110 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/pkg/river/ast"
+	"github.com/grafana/agent/pkg/river/diag"
+	"github.com/grafana/agent/pkg/river/vm"
+)
+
+const (
+	configNodeID = "configNode"
+
+	loggingBlockID = "logging"
+)
+
+// ConfigNode is a controller node which manages agent configuration.
+type ConfigNode struct {
+	mut          sync.RWMutex
+	blocks       []*ast.BlockStmt // Current River blocks to derive config from
+	logger       log.Logger
+	loggingArgs  logging.Options // Evaluated logging arguments for the config
+	loggingBlock *ast.BlockStmt
+	loggingEval  *vm.Evaluator
+}
```

Comment:
Is the intent that all of our config blocks are represented by one ConfigNode instance? Just want to check :)

Reply:
Yeah. Built this with the intention that there'd _always_  exactly one single ConfigNode instance on the graph.

Multiple config blocks would be stored in the slice, and each one would be used for specific subset of the struct fields. Not 100% sure about this approach yet, still playing around to see how multiple/none config nodes could work.